spec.md

```diff
@@ -2110,6 +2230,57 @@ Condition | gRPC Code | Description | Recovery Behavior
 | Call not implemented | 12 UNIMPLEMENTED | NodeGetInfo call is not implemented by the plugin or disabled in the Plugin's current mode of operation. | Caller MUST NOT retry. Caller MAY call `ControllerGetCapabilities` or `NodeGetCapabilities` to discover Plugin capabilities. |
 
 
+#### `NodeExpandVolume`
+
+A Node Plugin MUST implement this RPC call if it has `EXPAND_VOLUME` node capability.
+This RPC call allows CO to expand volume on a node.
+
+`NodeExpandVolume` ONLY supports expansion of already node published or node staged volumes on given `volume_path`.
+A plugin may return `FAILED_PRECONDITION` error if underlying volume can not be expanded when published or staged.
+
+Example:
+If plugin has `PUBLISH_UNPUBLISH_VOLUME` controller capability:
+* `NodeExpandVolume` MUST be called after calling `ControllerPublishVolume`.
+* If plugin has `STAGE_UNSTAGE_VOLUME` node capability then `NodeExpandVolume` MUST be called after `NodeStageVolume`.
+* `NodeExpandVolume` however MAY be called before or after `NodePublishVolume`.
+
+If plugin does not have `PUBLISH_UNPUBLISH_VOLUME` controller capability:
+* `NodeExpandVolume` MUST be called after `NodeStageVolume` if plugin has `STAGE_UNSTAGE_VOLUME` node capability.
+* `NodeExpandVolume` however MAY be called before or after `NodePublishVolume`.
+
+
+```protobuf
+message NodeExpandVolumeRequest {
+  // The ID of the volume. This field is REQUIRED.
+  string volume_id = 1;
+
+  // The path on which volume is available.
+  // This field is REQUIRED.
+  string volume_path = 2;
+
+  // This field is OPTIONAL. This allows CO to specify the
+  // capacity requirements of the volume after expansion.
+  CapacityRange capacity_range = 3;
+}
+
+message NodeExpandVolumeResponse {
+  // The capacity of the volume in bytes.
+  // This field is OPTIONAL.
+  int64 capacity_bytes = 1;
+}
+```
+
+##### NodeExpandVolume Errors
+
+| Condition             | gRPC code | Description           | Recovery Behavior                 |
+|-----------------------|-----------|-----------------------|-----------------------------------|
+| Volume in use | 9 FAILED_PRECONDITION | Indicates that the volume corresponding to the specified `volume_id` could not be expanded because it is in use. | Caller SHOULD ensure that volume is not in-use and retry with exponential back off. |
```

Comment:
Ditto. Let's better define "in use" like i suggested above.

Reply:
I am thinking of renaming this to:

```
Indicates that the volume corresponding to the specified `volume_id` could not be expanded because it is published or staged to the node and underlying filesystem does not support expansion of published or staged volumes. | Caller MUST not retry. |
```

Also plugin capabilities do not really apply to node.